### PR TITLE
chore(docs): rename @private to @internal

### PR DESCRIPTION
**Pull request by alexeagle**
The latter is understood by TypeScript's --stripInternal option, so this lets us
rely more on the tooling provided by typescript team.

Part of #3082

### CODE REVIEW

**Comment by alexeagle** [comment]
cc @IgorMinar @petebacondarwin I've tried to be careful in the docs generation code to replace the right `private` references, but I'm not confident how to test that the docs are generated the same way. Should I do more careful testing before merging this?

**Comment by petebacondarwin** [comment]
If you generate the docs in angular.io, both before and after your changes, then you should be able to see if there are any changes via git.

**Comment by petebacondarwin** [comment]
Hold on, we have migrated all these dgeni packages over to angular.io. You should be making changes there.

File: docs/typescript-package/services/convertPrivateClassesToInterfaces.spec.js [hunk 1]
```diff
@@ -11,13 +11,13 @@ describe('readTypeScriptModules', function() {
     convertPrivateClassesToInterfaces = injector.get('convertPrivateClassesToInterfaces');
   });
 
-  it('should convert private class docs to interface docs', function() {
+  it('should convert @internal class docs to interface docs', function() {
```
Comment (petebacondarwin) [inline docs/typescript-package/services/convertPrivateClassesToInterfaces.spec.js]: I am not sure if you wanted to make this change. This is about class that have private constructors

**Comment by IgorMinar** [comment]
as @petebacondarwin said this PR needs to be split into two. one with annotation changes in our code base and then the second one should be for angular.io repo to update the dgeni package.

**Comment by IgorMinar** [comment]
I'll send a PR removing the dgeni code from our angular/angular

**Comment by alexeagle** [comment]
the angular.io change is already merged actually (Naomi merged the PR a bit before I was ready...)

**Comment by alexeagle** [comment]
merged in f7aa890adef00de8dc2f421dd317220994ae4d20

**Pull request closed without merging**